### PR TITLE
KTOR-4960 Make sessions plugin and auth provider multiplatform

### PR DESCRIPTION
**Pull request by rsinukov**
https://youtrack.jetbrains.com/issue/KTOR-4960

No real changes, just moving and splitting files + couple of `expect`/`actual`

### CODE REVIEW

**Comment by e5l** [comment]
@rsinukov please check compilation

**Comment by rsinukov** [comment]
@e5l thanks, fixed!

File: ktor-server/ktor-server-plugins/ktor-server-sessions/jvmAndNix/src/io/ktor/server/sessions/Cache.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+package io.ktor.server.sessions
+
+import io.ktor.util.collections.*
+import kotlinx.coroutines.*
+import kotlin.coroutines.*
+
+@Suppress("KDocMissingDocumentation")
+public interface Cache<in K : Any, V : Any> {
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-sessions/jvmAndNix/src/io/ktor/server/sessions/Cache.kt:12]: Could you please add KDoc?

File: ktor-server/ktor-server-plugins/ktor-server-sessions/jvmAndNix/src/io/ktor/server/sessions/CacheStorage.kt [hunk 2]
```diff
@@ -5,9 +5,13 @@
 package io.ktor.server.sessions
 
 @Suppress("KDocMissingDocumentation")
-public class CacheStorage(public val delegate: SessionStorage, idleTimeout: Long) : SessionStorage {
-    private val referenceCache = SoftReferenceCache<String, String> { id -> delegate.read(id) }
-    private val cache = BaseTimeoutCache(idleTimeout, true, referenceCache)
+
+public class CacheStorage constructor(
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-sessions/jvmAndNix/src/io/ktor/server/sessions/CacheStorage.kt:9]: same here

**Review by e5l — APPROVED** [review]
LGTM in general

**Pull request merged**